client/state/themes/selectors.js

```diff
@@ -400,7 +400,7 @@ export function getThemeHelpUrl( state, theme, siteId ) {
  * @return {?String}        Theme purchase URL
  */
 export function getThemePurchaseUrl( state, theme, siteId ) {
-	if ( ! isPremiumTheme( theme ) ) {
+	if ( isJetpackSite( state, siteId ) || ! isThemePremium( state, theme.id ) ) {
```

Comment:
Not yet, at least. Not quite sure about how this is going to be handled with Pressable.